Add options to enable terminal bell on events

mphe opened the pull request:
Closes #342 .

Adds options to enable or disable the terminal bell on message, filetransfer and invite events.

mphe committed:
Add options to enable terminal bell on certain events

Some terminals can mark the terminal window as urgent on bell.
This is useful for window managers that provide a shortcut to jump to an
urgent client.

mphe committed:
Update docs

JFreegman commented:
We already have options to enable sound notifications for various types of events, which is the default behaviour. Now with these new options enabled on top of that, we end up with both sound notifications and a terminal bell for a given event, which is both confusing to config, and redundant for the user. The problem is that sound notifications don't work in certain circumstances (e.g. using toxic via ssh or putty), so the desired behaviour should be that it simply falls back to the terminal bell on the given events with no extra config required. We shouldn't need redundant options for sound alerts.

mphe commented:
No, it's not redundant, as this is not meant for sound notifications.

Terminals often set the window status to urgent on bell. Urgent windows usually appear highlighted in the taskbar and some window managers, like Awesome, even have shortcuts to jump to the next urgent client.

As already mentioned in #342, this is incredible useful if you have these mentioned keybindings and/or want to have a visual notification + sound notifications.

JFreegman commented:
Could you make the distinction more clear in the example and man pages then? First we have the `alerts` option that says: `Enable or disable terminal alerts on events`, then we have the new options that say `Beep when [event]`. It's unclear what the difference is, and how they relate to each other.

mphe committed:
Add further explanation to the beep_on* options

This also renames beep_on* to bell_on*.

mphe commented:
Ok, I added a bit more explanation on when the beep_on\* options are useful. I also renamed the beep_on\* options to bell_on\* so that it fits better in the context.
